packages/itwinui-react/src/core/utils/hooks/useStyles.ts

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect.js';
+import { getDocument } from '../functions/dom.js';
+import * as React from 'react';
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment -- only gets created at build-time
+// @ts-ignore
+import rawCssText from '../../../styles.js';
+
+// react <18 fallback for useInsertionEffect, with workaround for webpack getting rid of React namespace
+const _React = React;
+const useIsomorphicInsertionEffect =
+  _React.useInsertionEffect ?? useIsomorphicLayoutEffect;
+
+/**
+ * Dynamically loads the iTwinUI styles as a constructed stylesheet,
+ * falling back to a regular `<style>` tag in `<head>`.
+ */
+export const useStyles = (options?: {
+  withLayer?: boolean;
+  document?: () => Document | undefined;
+}) => {
+  const { withLayer = true, document = () => getDocument() } = options ?? {};
```

Comment:
by default, v2 _will_ be added to layer. you can inspect any of the components.

to turn layers off (in case something is breaking and impossible to fix), user can do this:
```jsx
<ThemeProvider styleOptions={{ withLayer: false }}>
```

(suggestions for better prop names welcome)

Reply:
Does it mean that users can cherry pick what components use which layer?

Reply:
no, it's all or nothing, since there is a single stylesheet.

Reply:
I was just thinking, if the current approach is no layer at all, then maybe `withLayer` prop should be 'false' by default. Adding styles to a layer if user does not expect layers could break things (I think).

Reply:
ok, makes sense